src/csharp/Microsoft.Spark.E2ETest/UdfTests/UdfSimpleTypesTests.cs

```diff
@@ -74,9 +92,74 @@ public void TestUdfWithReturnAsDateType()
             // Chained UDFs.
             {
                 Row[] rows = _df.Select(udf2(udf1(_df["age"]))).Collect().ToArray();
-                Assert.Equal(3, rows.Length);
+                Assert.Equal(2, rows.Length);
+
+                var expected = new string[] { "2020-01-04", "2050-01-04" };
+                for (int i = 0; i < rows.Length; ++i)
+                {
+                    Assert.Equal(1, rows[i].Size());
+                    Assert.Equal(expected[i], rows[i].GetAs<string>(0));
+                }
+            }
+        }
+
+        /// <summary>
+        /// UDF that takes in Timestamp type.
+        /// </summary>
+        [Fact]
+        public void TestUdfWithTimestampType()
+        {
+            Func<Column, Column> udf = Udf<Timestamp, string>(time => time.ToString());
+
+            Row[] rows = _df.Select(udf(_df["time"])).Collect().ToArray();
+            Assert.Equal(2, rows.Length);
+
+            var expected = new string[]
+            {
+                "2020-01-01 00:00:00.000000",
+                "2020-01-02 15:30:30.000000"
```

Comment:
I assume this is the same format that Scala/Python uses for toString?

Reply:
Yes, same format.